server/src/test/java/org/elasticsearch/index/RequiredPipelineIT.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index;
+
+import org.elasticsearch.action.index.IndexRequestBuilder;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ESIntegTestCase;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasToString;
+
+public class RequiredPipelineIT extends ESIntegTestCase {
```

Comment:
Should we extend from `ESSingleNodeTestCase` instead of `ESIntegTestCase`?
The former is more lightweight and we don't require multiple nodes in this test.

Reply:
Actually it was through the use of multiple nodes where sometimes the coordinating node is not an ingest node that I came to understand the need to mark the pipeline as resolved. I think that we should keep this as-is, since coordinating nodes that are not ingest nodes behave slightly differently.

Reply:
Sorry, I missed that the last test was actually sending an index request. In that case it makes sense to keep using the existing test base class.